src/vcpkg/commands.add-version.cpp

```diff
@@ -379,12 +472,21 @@ namespace vcpkg::Commands::AddVersion
 
         for (auto&& port_name : port_names)
         {
-            // Get version information of the local port
+            auto port_dir = paths.builtin_ports_directory() / port_name;
+
+            if (!fs.exists(port_dir, VCPKG_LINE_INFO))
+            {
+                msg::print_error(msgAddVersionPortDoesNotExist, msg::package_name = port_name);
+                if (add_all) continue;
+                Checks::exit_fail(VCPKG_LINE_INFO);
```

Comment:
This pattern being repeated so often strongly suggests that this could use an 'extract function' refactoring but no concrete change requested at this time.